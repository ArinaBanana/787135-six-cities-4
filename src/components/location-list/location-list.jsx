import React, {Component} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../store/actions/location";

import Location from "../location/location.jsx";

class LocationList extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {locations, currentLocation, onTitleClick} = this.props;

    return (
      <ul className="locations__list tabs__list">
        {
          locations.map((location, index) => {
            const isActive = location === currentLocation;

            return <Location
              key={`${index}-${location}`}
              city={location}
              isActive={isActive}
              onTitleClick={onTitleClick} />;
          })
        }
      </ul>
    );
  }
}

const mapStateToProps = (state) => ({
  locations: state.LOCATIONS.locations,
  currentLocation: state.LOCATIONS.currentLocation
});

const mapDispatchToProps = (dispatch) => ({
  onTitleClick(city) {
    dispatch(ActionCreator.setLocation(city));
  }
});

LocationList.propTypes = {
  locations: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  currentLocation: PropTypes.string.isRequired,
  onTitleClick: PropTypes.func.isRequired
};

export {LocationList};
export default connect(mapStateToProps, mapDispatchToProps)(LocationList);
