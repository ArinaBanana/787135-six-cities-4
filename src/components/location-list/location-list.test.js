import React from "react";
import renderer from "react-test-renderer";
import {LocationList} from "./location-list.jsx";

const locations = [`City`];

it(`Should render Location Lis`, () => {
  const tree = renderer.create(<LocationList
    locations={locations}
    currentLocation={`City`}
    onTitleClick={() => {}} />).toJSON();

  expect(tree).toMatchSnapshot();
});
