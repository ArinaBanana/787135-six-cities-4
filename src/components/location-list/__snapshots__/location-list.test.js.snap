// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Location Lis 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
    onClick={[Function]}
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="#"
    >
      <span>
        City
      </span>
    </a>
  </li>
</ul>
`;
